crud/src/main/java/com/redhat/lightblue/mediator/Mediator.java

```diff
@@ -701,28 +721,32 @@ public BulkResponse bulkRequest(BulkRequest requests) {
                 if (requests.isOrdered()) {
                     // ordered - only consecutive finds in parallel
                     if (req.getOperation() == CRUDOperation.FIND) {
-                        ctx.futures[i] = executor.submit(getFutureRequest(req));
+                        ctx.futures[i] = executor.submit(getFutureRequest(req, metrics));
                     } else {
                         wait(ctx);
                         switch (req.getOperation()) {
                             case INSERT:
-                                ctx.setResponseAt(i, insert((InsertionRequest) req));
+                            	requestCtx = metrics.startEntityRequest("insert", req.getEntityVersion().getEntity(), req.getEntityVersion().getVersion());
```

Comment:
So per suggestion above, I would probably start and stop tracking here. This way we don't have to pass around the metricCtx everywhere, and we could do the same thing in REST layer, consistent with other requests.